msc_pygeoapi/loader/cap_alerts.py

```diff
@@ -0,0 +1,577 @@
+                    }
+                }
+            }
+        }
+    }
+}
+
+
+class CapAlertsRealtimeLoader(BaseLoader):
+    """Cap Alerts real-time loader"""
+
+    def __init__(self, plugin_def):
+        """initializer"""
+
+        BaseLoader.__init__(self)
+
+        self.ES = get_es(MSC_PYGEOAPI_ES_URL, MSC_PYGEOAPI_ES_AUTH)
+
+        if not self.ES.indices.exists(INDEX_NAME,
+                                      request_timeout=MSC_PYGEOAPI_ES_TIMEOUT):
+            self.ES.indices.create(index=INDEX_NAME, body=SETTINGS,
+                                   request_timeout=MSC_PYGEOAPI_ES_TIMEOUT)
+
+    def load_data(self, filepath):
+        """
+        fonction from base to load the data in ES
+
+        :param filepath: filepath for parsing the current condition file
+
+        :returns: True/False
+        """
+
+        data = self.weather_warning2geojson(filepath)
+
+        try:
+            self.bulk_data = []
+            for doc in data:
+                op_dict = {
+                    'index': {
+                        '_index': INDEX_NAME,
+                        '_type': '_doc'
+                    }
+                }
+                op_dict['index']['_id'] = doc['properties']['identifier']
+                self.bulk_data.append(op_dict)
+                self.bulk_data.append(doc)
+            r = self.ES.bulk(index=INDEX_NAME, body=self.bulk_data)
+
+            LOGGER.debug('Result: {}'.format(r))
+
+            previous_alerts = self.delete_references_alerts()
+
+            click.echo('done importing in ES')
+
+            if previous_alerts:
+                LOGGER.debug('Deleted old warning')
+            else:
+                LOGGER.debug('New warning, no deletion')
+            return True
+
+        except Exception as err:
+            LOGGER.warning('Error bulk indexing: {}'.format(err))
+            return False
+
+    def delete_references_alerts(self):
+        """Delete old alerts documents"""
+
+        if self.references_arr and len(self.references_arr) != 0:
+
+            es = get_es(MSC_PYGEOAPI_ES_URL, MSC_PYGEOAPI_ES_AUTH)
+
+            click.echo('Deleting old alerts')
+
+            query = {
+                'query': {
+                    'terms': {
+                        'properties.reference': self.references_arr
+                    }
+                }
+            }
+
+            es.delete_by_query(index=INDEX_NAME, body=query)
+
+            return True
+
+        else:
+            return False
+
+    def _get_date_format(self, date):
+        """
+        Convenience function to parse CAP dates
+
+        :param date: date form the cap XML
+
+        returns: date as datetime object
+        """
+        for char in ["T", "-", ":"]:
+            if char in date:
+                date = date.replace(char, '')
+        date = date[:14]
+        date = datetime.strptime(date, "%Y%m%d%H%M%S")
+
+        return date
+
+    def _get_element(self, node, path, attrib=None):
```

Comment:
Maybe move this into [msc_pygeoapi/util.py](https://github.com/ECCC-MSC/msc-pygeoapi/blob/master/msc_pygeoapi/util.py) since it's something that we can use across various loaders?

Reply:
done, thanks for the improvment :)

Reply:
ha moved the _get_date_format function :p we should also move the _get_element one ?

Reply:
makes sense.

Reply:
also moved _get_element to utils

Reply:
Awesome, all good on my end!